File: zad3/zadanie3.cpp
#include <iostream>
# include <vector>

using namespace std;

int countElements( const vector<int>& massiv);
void zapolnenie(int n , vector<int>& massiv);
void printVector(const vector<int>& massiv);

int main(){
    int n {};
    vector<int> massiv;
    cout<<"Введите N\n";
    cin>> n;
    cout<< "Введите массив до N\n";
    zapolnenie(n, massiv);
    cout<<"Введенный массив\n";
    printVector(massiv);
    cout<< "\n Колличество элементов равному предыдущему числу : ";
    cout << countElements(massiv)<<endl;
    return 0;
}

int countElements( const vector<int>& massiv){
    int count {};
    for (size_t i = 1; i < massiv.size() ; ++i){
        if (massiv[i] == massiv[i-1]){
            ++count;
        }
    }
    return count;
}
void zapolnenie(int n , vector<int>& massiv){
    int numbers{};
    for (int i{}; i <n ; ++i){
        cin>> numbers;
        massiv.push_back(numbers);
    }
}
void printVector(const vector<int>& massiv){
    for (int val : massiv){
        cout<< val << " ";
    }
}

File: zad3/zadanie3.ex
defmodule Counter do
  def main do
    IO.puts("Введите N")
    n = IO.gets("") |> String.trim() |> String.to_integer()
    IO.puts("Введите массив до N")
    numbers = read_numbers(n, [])
    IO.puts("Введенный массив")
    IO.puts(Enum.join(numbers, " "))
    result = count_elements(numbers)
    IO.puts("\nКоличество элементов, равных предыдущему: #{result}")
  end

  defp read_numbers(0, acc), do: Enum.reverse(acc)
  defp read_numbers(n, acc) do
    line = IO.gets("")
    case line do
      nil -> read_numbers(n, acc)
      str ->
        case Integer.parse(String.trim(str)) do
          {num, ""} -> read_numbers(n - 1, [num | acc])
          _ -> read_numbers(n, acc)
        end
    end
  end

  defp count_elements([]), do: 0
  defp count_elements([_]), do: 0
  defp count_elements([first, second | rest]) do
    count = if first == second, do: 1, else: 0
    count + count_elements([second | rest])
  end
end

Counter.main()
